Makefile

```diff
@@ -81,56 +105,57 @@ ROOT := $(shell pwd)
 
 # MAKE PROJECT
 
-.PHONY: all lint proj program
+.PHONY: all lint project program gdb
 
 # Actually calls the make
 all: project lint
 
-# Includes device specific configurations
-include device/$(DEVICE_FAMILY)/device_config.mk
+# Includes platform-specific configurations
+-include $(PLATFORMS_DIR)/$(PLATFORM)/platform.mk
 
 # Includes all libraries so make can find their targets
 $(foreach dep,$(LIBS),$(call include_lib,$(dep)))
 
-# Lints the files in ms-lib and projects
+# Lints the files in ms-common and projects
 lint:
-	@-find projects -name "*.c" -o -name "*.h" | xargs -P 24 -r python2 lint.py
-	@-find libraries -path "$(LIB_DIR)/stm32f0xx" -prune -o -name "*.c" -o -name "*.h" | xargs -P 24 -r python2 lint.py
+	@find $(PROJECTS_DIR) -name "*.c" -o -name "*.h" | xargs -P 24 -r python2 lint.py
+	@find "$(LIB_DIR)/ms-common" -name "*.c" -o -name "*.h" | xargs -P 24 -r python2 lint.py
 
 # Builds the project
-project: $(BIN_DIR)/$(PROJECT_NAME).elf
+project: $(BIN_DIR)/$(PROJECT)$(PLATFORM_EXT)
 
 # Rule for making the project
-$(BIN_DIR)/%.elf: $(MAIN_FILE) $(HEADERS) $(STARTUP) $(APP_LIBS) | $(BIN_DIR)
-	@$(CC) $(CFLAGS) $^ -o $@ -L$(STATIC_LIB_DIR)\
-		$(foreach dep,$(APP_DEPS), -l$(notdir $(dep))) \
-		$(LINKER)
-	@$(OBJCPY) -O binary $@ $(BIN_DIR)/$(PROJECT_NAME).bin
+$(BIN_DIR)/%$(PLATFORM_EXT): $(MAIN_FILE) $(APP_LIBS) | $(BIN_DIR)
+	@$(CC) $(CFLAGS) $^ -o $@ -L$(STATIC_LIB_DIR) \
+		$(addprefix -l,$(APP_DEPS)) \
+		$(LDFLAGS) $(addprefix -I,$(INC_DIRS))
+	@$(OBJCPY) -O binary $@ $(BIN_DIR)/$(PROJECT).bin
 	@$(OBJDUMP) -St $@ >$(basename $@).lst
 	$(SIZE) $@
 
 $(DIRS):
 	@mkdir -p $@
 
-# OPTIONAL:
-# $(OBJCPY) -o ihex $(PROJECT_NAME).hex
-
 ###################################################################################################
 
 # OPENOCD SUPPORT
 
-# TODO verify this isn't broken
-program: $(BIN_DIR)/$(PROJECT_NAME).bin
+program: $(BIN_DIR)/$(PROJECT).bin
```

Comment:
Not sure why but this doesn't seem to recompile the program. I made a change and ran program it didn't work. Altered nothing and ran gdb it worked. Ran program again and it worked.